### PR TITLE
Added sticky settings functionality to flyout

### PR DESCRIPTION
**Pull request by Blackbaud-TrevorBurch**

### CODE REVIEW

File: src/app/public/modules/flyout/flyout.component.ts [hunk 1]
```diff
@@ -447,6 +464,23 @@ export class SkyFlyoutComponent implements OnDestroy, OnInit {
...
   }
 
+  private setUserData() {
+    if (this.config.settingsKey) {
+      this.uiConfigService.setConfig(
+        this.config.settingsKey,
+        {
+          flyoutWidth: this.flyoutWidth
+        }
+      ).subscribe(
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/flyout/flyout.component.ts]: Needs a `takeUntil()`.

File: src/app/public/modules/flyout/flyout.component.spec.ts [hunk 2]
```diff
@@ -443,6 +508,47 @@ describe('Flyout component', () => {
     expect(mouseUpSpy).toHaveBeenCalled();
   }));
 
+  it('should send the resized width to the UI config service when a settings key is given', fakeAsync(() => {
+    openFlyout({ defaultWidth: 500, settingsKey: 'testKey' });
+    fixture.detectChanges();
+    tick();
+    spyOn(SkyFlyoutComponent.prototype, 'onMouseMove').and.callThrough();
+    spyOn(SkyFlyoutComponent.prototype, 'onHandleRelease').and.callThrough();
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/flyout/flyout.component.spec.ts]: Why are you spying on these two mouse events? I don't see you utilizing them anywhere...

File: src/app/public/modules/flyout/flyout.component.ts (hunk 1 above)
Comment (Blackbaud-TrevorBurch) [reply src/app/public/modules/flyout/flyout.component.ts]: Added a `take(1)`

File: src/app/public/modules/flyout/flyout.component.spec.ts (hunk 2 above)
Comment (Blackbaud-TrevorBurch) [reply src/app/public/modules/flyout/flyout.component.spec.ts]: There were two tests like this due to copy pasta. I've fixed them both.

**Comment by Blackbaud-TrevorBurch** [comment]
@Blackbaud-AlexKingman ready for you again. I changed how min and max detection happens so that we no longer check for that until after we grab the sticky settings. There was also some bugginess with timing with the sticky settings - these changes fix that.

**Pull request merged**